### PR TITLE
Add stencil chain apps

### PR DESCRIPTION
**Pull request by psuriana**

### CODE REVIEW

**Comment by abadams** [comment]
We don't want this to break, so we should probably add a test target to the makefile, add it to make test_apps in the top-level makefile, and also add a CMakeLists.txt file so that it's buildable with cmake. (copy one from one of the other apps).

The test target should just run the benchmark. No need to actually test anything.

File: apps/stencil_chain/stencil_chain_generator.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            }
+            f(x, y) = e;
+            stages.push_back(f);
+        }
+
+        output(x, y) = stages.back()(x, y);
+
+        if (auto_schedule) {
+            // Provide estimates on the input image
+            input.dim(0).set_bounds_estimate(0, 1536);
```
Comment (dsharletg) [inline apps/stencil_chain/stencil_chain_generator.cpp]: Use constant variable for width and height

File: apps/stencil_chain/process.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    // Manually-tuned version
+    double best_manual = benchmark(timing, 1, [&]() {
+        stencil_chain(input, output);
+    });
+    printf("Manually-tuned time: %gms\n", best_manual * 1e3);
+
+    #ifndef NO_AUTO_SCHEDULE
+    // Auto-scheduled version
+    double best_auto = benchmark(timing, 1, [&]() {
+        stencil_chain(input, output);
```
Comment (dsharletg) [inline apps/stencil_chain/process.cpp]: Looks like this benchmarks the same thing as above?

**Comment by psuriana** [comment]
@abadams PTAL

**Comment by abadams** [comment]
I think this app has been edited a bunch in the autoschedule_with_convnet branch. You may want to pull the changes from there.

**Comment by psuriana** [comment]
I've merged the changes from convnet branch. @abadams  PTAL

**Pull request merged**